### DISCUSSION TITLE
Github Copilot Business w/proxy and self-signed certificates

### DISCUSSION THREAD

**Discussion by santif** [mitmproxy/mitmproxy#6067] [Intercept Everything!]
I am using Github Copilot Business for research purposes and need to collect telemetry data sent to Github servers. To achieve this, I'm attempting to configure Copilot with a local proxy. However, when I try to use mitmproxy, I encounter an error in VSCode: "Your current Copilot license does not support proxy connections with self-signed certificates."

I have referred to Github Copilot's documentation, specifically the troubleshooting guide for certificate errors (https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-certificate-errors-for-github-copilot), but have been unable to find a configuration option to do so.

**Comment by mhils** [comment]
This sounds like a Copilot issue, not a mitmproxy bug?

**Reply by santif** [reply]
Indeed, this could be seen as a limitation or bug of Copilot. However, since one of the known workarounds is to configure a proxy that "does not intercept secure connections", I was wondering if any suitable configuration could be applied in mitmproxy. Taking advantage of the fact that mitmproxy looks like the tool that I exactly need to advance my research on Copilot's telemetry.

**Comment by jbochi** [comment]
I was able to use mitmproxy to inspect the requests the VS Code extension makes.

I started the proxy like this: `mitmproxy --mode reverse:https://copilot-proxy.githubusercontent.com`

And I included this in `settings.json`:

```
    "github.copilot.advanced": {
        "debug.overrideProxyUrl": "http://localhost:8080"
    },
```

**Comment by monperrus** [comment]
thanks a lot @jbochi it works :+1:
